open-company/open-company-web
# Draft to new section

Card: https://trello.com/c/JvEN5yD0

Related PR:
- storage https://github.com/open-company/open-company-storage/pull/131

Solves 2 major issues: remove a section that contains drafts and add a draft to a new section.

This introduce a new flag for sections: draft. This flag is set to true if the current section has only draft entries in it. It means the section should be hidden from the left navigation bar of the web app but still be visible when picking a section in edit mode.

NB: a draft board is deleted when deleting/moving an entry inside it when:
1) it's marked as draft
2) it's empty
A section for being marked as draft has to be:
1) created from within a draft (not directly from the UI)
2) created in any way but deleted by the user: if the section has drafts inside it's marked as draft instead of deleted
This is why we can delete a draft board when it's empty.

Test 1:
- [x] tests in the trello card

Test 2:
- click Compose (+ New)
- compose a draft and add it to a new section
- save as draft
- [x] do you NOT see the new section in left nav? Good
- [x] do you see the new section as a possible section in edit mode? Good
- now compose again
- compose a post to the section above
- post it!
- [x] do you see the section in the left nav now? Good
- [x] do you see the old draft in drafts? Good
- delete the section now
- [x] do you NOT see the section in left nav? Good
- [x] do you NOT see the post published in this section in AP? Good
- [x] do you still see the draft in drafts? Good
- delete the draft
- refresh
- [x] do you NOT see the section in left nav? Good
- compose
- [x] do you still see the section in the picker? Good
- publish a post to it
- click section settings
- change to public
- share automatically to slack channel
- save
- delete the section
- [x] do you still see the draft? Good
- compose
- [x] can you publish to the section? Good
- publish a post to the section
- click settings
- [x] section settings where preserved? Good
- now publish all the remaining drafts of this section
- [x] delete the section again
- [x] do you NOT see the section in left nav? Good
- [x] do you NOT see the section when composing? Good
